.github/workflows/tests-ipsec-upgrade.yaml

```diff
@@ -187,9 +188,20 @@ jobs:
             # "-ci" suffix
             IMAGE_TAG=''
           fi
+          echo "CILIUM_DOWNGRADE_VERSION: ${CILIUM_DOWNGRADE_VERSION}"
+          echo "IMAGE_TAG: ${IMAGE_TAG}"
+          if [ -z "${CILIUM_DOWNGRADE_VERSION}" ]; then
+            echo "::notice::No CILIUM_DOWNGRADE_VERSION returned; skipping remaining steps"
+          fi
           echo downgrade_version=${CILIUM_DOWNGRADE_VERSION} >> $GITHUB_OUTPUT
           echo image_tag=${IMAGE_TAG} >> $GITHUB_OUTPUT
 
+      - name: Call actions/checkout again to remove credentials
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ inputs.context-ref || github.sha }}
+          persist-credentials: false
```

Comment:
My understanding is that it unsets the `http.https://github.com/.extraheader` Git config option used for passing the token with the requests (and the `core.sshCommand` as well, that we're not using as far as I understand), so I'd expect them to gone from the repo? I'm not sure how to check this for good :thinking: 

![Screenshot_20240221_183127](https://github.com/cilium/cilium/assets/17001771/8cca262c-d7af-47a7-8121-f8915de5f42f)

Reply:
Good idea, I added it as a failsafe: https://github.com/cilium/cilium/pull/30742/files#diff-07b1303f71b74ecfe10ad34472da7c7e9b79ac9274fd93fe833ecc1551898473R204 and tested it successfully.

Reply:
Except that, obviously, `git fetch` is _supposed_ to work on `cilium/cilium` where we don't need any auth to fetch the code :facepalm:. I need to implement it differently.

Reply:
Added a check as follows:

```yaml
      - name: Check we effectively removed Git credentials
        shell: bash
        run: |
          if ! curl -sL \
            -H "Accept: application/vnd.github+json" \
            -H "X-GitHub-Api-Version: 2022-11-28" \
            "https://api.github.com/repos/${{ github.repository }}" | \
            jq --exit-status '.private == false'; then
            echo 'Checking whether "git fetch" succeeds'
            if git fetch origin HEAD; then
              echo "::error::Git credentials not removed, aborting now."
              false
            fi
          fi
```

Reply:
> Emphasizing that this check can return false positives in case of network errors

Sorry, I'm only seeing your comment now. Hmm yep, network errors would make the test “pass” as well. Not sure what we can do about it, maybe parsing the command output but that's a check on a check on a check and looks brittle...

My current implementation also has another limitation, which is that it doesn't run this check on a public repo (such as `cilium/cilium`), because I don't know how to test, without pushing stuff, that auth has been removed in that case :disappointed:

Reply:
I discussed offline with Feroz. Not sure there's much to do for either case, the network errors or the workflow skipping the check on `cilium/cilium`. As far as I understand, the token we use only has limited permissions anyway (the only thing that regular public users couldn't do without it is setting the status for the workflow), so it seems sufficient to keep the check as is.

I'll go ahead and send to the merge queue so that patch releases for March don't get blocked on this.

Thanks all!